pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -52,9 +48,53 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
-	// TODO(user): Modify this to be the types you create that are owned by the primary resource
-	// Watch for changes to secondary resource Pods and requeue the owner UnifiedPushServer
-	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
+	// Watch for changes to secondary resource Deployment and requeue the owner UnifiedPushServer
+	err = c.Watch(&source.Kind{Type: &appsv1.Deployment{}}, &handler.EnqueueRequestForOwner{
```

Comment:
To keep easier understand the code and keep it maintained I'd recommend following the approach adopted [here](https://github.com/aerogear/mobile-security-service-operator/tree/master/pkg/controller/mobilesecurityserviceapp)